homeassistant/components/devolo_home_control/switch.py

```diff
@@ -0,0 +1,161 @@
+"""Platform for light integration."""
+import logging
+
+from devolo_home_control_api.homecontrol import get_sub_device_uid_from_element_uid
+
+from homeassistant.components.switch import SwitchDevice
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.helpers.typing import HomeAssistantType
+
+from .const import DOMAIN
+
+logging.basicConfig(level=logging.DEBUG)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistantType, entry: ConfigEntry, async_add_entities
+) -> None:
+    """Get all devices and setup the switch devices via config entry."""
+    devices = hass.data[DOMAIN]["homecontrol"].binary_switch_devices
+
+    devices_list = []
+    for device in devices:
+        for i in range(len(device.binary_switch_property)):
+            devices_list.append(
+                DevoloSwitch(
+                    hass=hass,
+                    device_instance=device,
+                    sub_uid=get_sub_device_uid_from_element_uid(
+                        [*device.binary_switch_property][i]
+                    ),
+                )
+            )
+    async_add_entities(devices_list, False)
+
+
+class DevoloSwitch(SwitchDevice):
+    """Representation of an Awesome Light."""
+
+    def __init__(self, hass, device_instance, sub_uid):
+        """Initialize an devolo Switch."""
+        self._device_instance = device_instance
+
+        # Create the unique ID
+        if sub_uid is not None:
+            self._unique_id = self._device_instance.uid + "#" + str(sub_uid)
+        else:
+            self._unique_id = self._device_instance.uid
+
+        self._homecontrol = hass.data[DOMAIN]["homecontrol"]
+        self._name = self._device_instance.itemName
+        self._available = self._device_instance.is_online()
+
+        # Get the brand and model information
+        try:
+            self._brand = self._device_instance.brand
+            self._model = self._device_instance.name
+        except AttributeError:
```

Comment:
Why would there be `AttributeError`? All instance attributes should be set in init of an instance, with at least defaults.

Reply:
We do a call against a z-wave device Api, which sometimes do not return values for the requested devices. I would need to extend the api to catch this.

Reply:
Ok.

Reply:
Please make a comment about this for future readers.

Reply:
We will extend the api to have this value always with a default value. Will remove this try except here.

Reply:
removed this try except due to the fact that the api has defaults values for this now